bubuku/features/rebalance_by_size.py

```diff
@@ -0,0 +1,116 @@
+import json
+import logging
+
+from kazoo.exceptions import NodeExistsError
+
+from bubuku.broker import BrokerManager
+from bubuku.controller import Check, Change
+from bubuku.utils import CmdHelper
+from bubuku.zookeeper import Exhibitor
+
+_LOG = logging.getLogger('bubuku.features.rebalance_by_size')
+
+
+class RebalanceBySizeChange(Change):
+    def __init__(self, zk: Exhibitor):
+        self.zk = zk
+
+    def get_name(self):
+        return 'rebalance_by_size'
+
+    def can_run(self, current_actions):
+        raise NotImplementedError('Not implemented')  # todo: not implemented yet
+
+    def run(self, current_actions):
+        raise NotImplementedError('Not implemented')  # todo: not implemented yet
+
+    def can_run_at_exit(self):
+        raise NotImplementedError('Not implemented')  # todo: not implemented yet
+
+
+class RebalanceBySize(Check):
+    def __init__(self, zk: Exhibitor, broker: BrokerManager):
+        super().__init__(check_interval_s=600)
+        self.zk = zk
+        self.broker = broker
+
+    def check(self):
+        if self.broker.is_running_and_registered() and self.__is_data_imbalanced():
+            _LOG.info("Starting rebalance by size")
+            return RebalanceBySizeChange(self.zk)
+        return None
+
+    def __is_data_imbalanced(self) -> bool:
+        return False  # todo: not implemented yet
+
+
+class GenerateDataSizeStatistics(Check):
+    def __init__(self, zk: Exhibitor, broker: BrokerManager, cmd_helper: CmdHelper, kafka_log_dirs):
+        super().__init__(check_interval_s=1800)
+        self.zk = zk
+        self.broker = broker
+        self.cmd_helper = cmd_helper
+        self.kafka_log_dirs = kafka_log_dirs
+
+    def check(self):
+        if self.broker.is_running_and_registered():
+            _LOG.info("Generating data size statistics")
+            try:
+                self.__generate_stats()
+                _LOG.info("Data size statistics successfully written to zk")
+            except Exception:
+                _LOG.warn("Error occurred when collecting size statistics", exc_info=True)
+        return None
+
+    def __generate_stats(self):
+        topic_stats = self.__get_topics_stats()
+        disk_stats = self.__get_disk_stats()
+        stats = {"disk": disk_stats, "topics": topic_stats}
+        self.__write_stats_to_zk(stats)
+
+    def __get_topics_stats(self):
+        topic_stats = {}
+        for log_dir in self.kafka_log_dirs.split(","):
+            topic_dirs = self.cmd_helper.cmd_run("du -k -d 1 {}".format(log_dir)).split("\n")
+            for topic_dir in topic_dirs:
+                self.__parse_dir_stats(topic_dir, log_dir, topic_stats)
```

Comment:
I don't like the idea of having method that do not return result in this case...